goqtdrv5/cdrv.h

```diff
@@ -217,7 +222,69 @@ inline QByteArray drvGetByteArray(void *param)
         return QByteArray();
     }
     slice_head *sh = (slice_head*)param;
-    return QByteArray(sh->data,sh->len);
+    return QByteArray(sh->data, sh->len);
+}
+
+inline void drvSetByteArray(void *param, const QByteArray& array)
+{
+    if (param == 0) {
+        return;
+    }
+
+    slice_head *sh = (slice_head*)param;
+    sh->len = array.size();
+    sh->data = array.data();
+}
+
+inline QStringList drvGetStringArray(void *param)
```

Comment:
I tried that first, but it doesnt work. The only thing that would make it
work was to manually increment the pointer by 16. When doing it that way
the addresses were incorrect and the only the first string was being read
correctly. I think it was because the difference in address between the
array elements is different than with the go slice elements. Does that make
sense?
On Mon, Sep 7, 2015 at 5:36 PM Utkan Güngördü <notifications@github.com>
wrote:

> In goqtdrv5/cdrv.h
> <https://github.com/salviati/go-qt5/pull/26#discussion_r38880397>:
>
> > -    return QByteArray(sh->data,sh->len);
> > +    return QByteArray(sh->data, sh->len);
> > +}
> > +
> > +inline void drvSetByteArray(void *param, const QByteArray& array)
> > +{
> > +    if (param == 0) {
> > +        return;
> > +    }
> > +
> > +    slice_head *sh = (slice_head*)param;
> > +    sh->len = array.size();
> > +    sh->data = array.data();
> > +}
> > +
> > +inline QStringList drvGetStringArray(void *param)
>
> How about something along the lines of
>
> slice_head *sh = (slice_head*)param;
> QStringList stringList;
> string_head *goStrings = (string_head*)sh->data;
> for (goInt i=0; i<sh->len; i++) {
>     QString str = QString::fromUtf8(goStrings[i]->data, goStrings[i]->len);
>     stringList.push_back(str);
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/salviati/go-qt5/pull/26/files#r38880397>.
>